service-base/src/main/java/org/eclipse/hono/service/auth/device/X509AuthProvider.java

```diff
@@ -59,11 +59,21 @@ public X509AuthProvider(final CredentialsClientFactory credentialsClientFactory,
      * The JSON object passed in is required to contain a <em>subject-dn</em> property.
      * If the <em>singleTenant</em> service config property is {@code false}, then the
      * object is also required to contain a <em>tenant-id</em> property, otherwise
-     * the default tenant is used.
+     * the default tenant is used. If the authentication information contains a 
+     * <em>client-certificate</em> property, it is put into the returned credentials.
+     * <p>
+     * The values of the named properties are required to have the following types:
+     * <ul>
+     *     <li>tenant-id: {@link String}</li>
+     *     <li>subject-dn: {@link String}</li>
+     *     <li>client-certificate: Base64 encoded byte array as returned by {@link JsonObject#put(String, byte[])}</li>
+     * </ul>
```

Comment:
FMPOV there is no reason to restrict the additional properties to just *client-certificate* here.
We can allow any other properties in addition to *tenant-id* and *subject-dn* and simply copy all of them to the client context. WDYT?